### PR TITLE
fix #6271 use genesis dpa output file, convert genesis to elegant/opal

### LINKED ISSUE #6271: Add genesis back to elegant/opal workflow
Genesis can generate a DPA output file (enabled with IDMPPAR = 1) which can then be converted into particle info with longitudinal position using this library:

https://github.com/ChristopherMayes/openPMD-beamphysics/blob/master/pmd_beamphysics/interfaces/genesis.py#L146

### PR DESCRIPTION
**Pull request by moellep**

### CODE REVIEW

File: sirepo/package_data/template/omega/parameters.py.jinja [hunk 1]
```diff
@@ -80,31 +110,58 @@ def run_genesis(run_dir, genesis_id, prev_sim=None):
...
+            with open(lib_file, 'wb') as f:
+                d.tofile(f)
+        else:
+            if prev_sim.sim_type == "elegant":
+                p = switchyard.read_elegant(f"{prev_sim.outfile_path}")
+            else:
+                assert prev_sim.sim_type == "opal"
+                p = switchyard.read_opal(f"{prev_sim.outfile_path}")
+            particle_data = numpy.zeros([6, len(p.x)])
+            for i, col in enumerate(['pt', 'ct', 'x', 'y', 'ux', 'uy']):
```
Comment (gurhar1133) [inline sirepo/package_data/template/omega/parameters.py.jinja:132]: maybe store `['pt', 'ct', 'x', 'y', 'ux', 'uy']` in a constant that comes from the template?

File: sirepo/package_data/template/omega/parameters.py.jinja [hunk 2]
```diff
@@ -136,48 +193,33 @@ def sim_lib_file(sim_type, model, field, filename):
...
-                    text += f"? CHARGE = {charge}\n"
-                    continue
-            text += line
-    if charge:
-        pkio.write_text(lib_file, text)
+    io.partfile = filename
+    io.ippart = 0
+    io.ipradi = 0
+    d = numpy.fromfile(lib_file, dtype=numpy.float64)
+    n = len(d) / 6
```
Comment (gurhar1133) [inline sirepo/package_data/template/omega/parameters.py.jinja:215]: a lot of reshaping with 6 as a dim. Does 6 have a meaning that could be stored in var/constant name?

**Review by gurhar1133 — COMMENTED** [review]
Left a few comments. Not crucial stuff. Is the best way to test this just run a chain of sims where genesis comes before elegant/opal?

If so, do you have a good example chain of sims I should try?

**Comment by moellep** [comment]
Basically test if you can chain together sims which come from genesis and then go to either opal or elegant. You could also try genesis to genesis. Simple sims are elegant: Compact Storage Ring, opal: CSR Bend Drift, genesis: Pegasus FEL

**Comment by gurhar1133** [comment]
Works well lgtm

**Comment by moellep** [comment]
I've converted this PR to draft until pmd_beamphysics is available in sirepo common.

File: sirepo/package_data/template/omega/parameters.py.jinja [hunk 3]
```diff
@@ -36,6 +37,31 @@ def file_name_from_sim_name(sim_type, sim_name):
     return f"{res}.{ext}"
 
 
+def genesis_to_pmd(sim):
+    import pmd_beamphysics.interfaces.genesis
+    import pmd_beamphysics.particles
+
+    g = read_sim("genesis", sim.sim_id)
+    npart = g.models.electronBeam.npart
+    curpeak = g.models.electronBeam.curpeak
```
[the remark this answers is not in the document]
Comment (moellep) [reply sirepo/package_data/template/omega/parameters.py.jinja]: Removed temp variables with 536664a

File: sirepo/package_data/template/omega/parameters.py.jinja (hunk 2 above)
Comment (moellep) [reply sirepo/package_data/template/omega/parameters.py.jinja]: Replace 6 with constant in 536664a

File: sirepo/package_data/template/omega/parameters.py.jinja (hunk 1 above)
Comment (moellep) [reply sirepo/package_data/template/omega/parameters.py.jinja:129]: These items are the mapping from switchyard to genesis, but there isn't a full definition for it outside this file.

**Pull request merged**